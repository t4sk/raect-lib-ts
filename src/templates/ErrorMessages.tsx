import React from "react";
import { Transition, Message } from "semantic-ui-react";

interface Props {
  errors: string[];
}

export const ErrorMessages: React.FC<Props> = ({ errors }) => {
  return (
    <Transition
      visible={errors.length > 0}
      animation="fade down"
      duration={500}
    >
      {errors.length === 1 ? (
        <Message error={true}>{errors[0]}</Message>
      ) : (
        <Message error={true} list={errors} />
      )}
    </Transition>
  );
};

export default ErrorMessages;
