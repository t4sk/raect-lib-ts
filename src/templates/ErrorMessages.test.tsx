import React from "react";
import { render } from "@testing-library/react";
import { ErrorMessages } from "./ErrorMessages";

const props = {
  errors: [],
};

test("no error", () => {
  const { container } = render(<ErrorMessages {...props} />);

  expect(container).toMatchSnapshot();
});

test("one error", () => {
  const { container } = render(<ErrorMessages {...props} errors={["error"]} />);

  expect(container).toMatchSnapshot();
});

test("many errors", () => {
  const { container } = render(
    <ErrorMessages {...props} errors={["error 1", "error 2"]} />
  );

  expect(container).toMatchSnapshot();
});
