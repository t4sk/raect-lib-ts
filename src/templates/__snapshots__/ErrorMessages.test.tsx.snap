// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`many errors 1`] = `
<div>
  <div
    class="ui error message fade down visible transition"
  >
    <div
      class="content"
    >
      <ul
        class="list"
      >
        <li
          class="content"
        >
          error 1
        </li>
        <li
          class="content"
        >
          error 2
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`no error 1`] = `<div />`;

exports[`one error 1`] = `
<div>
  <div
    class="ui error message fade down visible transition"
  >
    error
  </div>
</div>
`;
